src/CSnakes.Runtime/Python/Internals/PyObjectImporters.cs

```diff
@@ -0,0 +1,153 @@
+using CSnakes.Runtime.CPython;
+
+namespace CSnakes.Runtime.Python.Internals;
+
+/// <summary>
+/// This type and its members, although technically public in visibility, are
+/// not intended for direct consumption in user code. They are used by the
+/// generated code and may be modified or removed in future releases.
+/// </summary>
+public interface IPyObjectImporter<out T>
+{
+    static abstract T Import(PyObject obj);
+}
+
+/// <summary>
+/// This type and its members, although technically public in visibility, are
+/// not intended for direct consumption in user code. They are used by the
+/// generated code and may be modified or removed in future releases.
+/// </summary>
+public static partial class PyObjectImporters
+{
+    public sealed class Runtime<T> : IPyObjectImporter<T>
+    {
+        public static T Import(PyObject obj) => obj.As<T>();
+    }
+
+    public sealed class Boolean : IPyObjectImporter<bool>
+    {
+        public static bool Import(PyObject obj) => CPythonAPI.IsPyTrue(obj);
```

Comment:
This requires the GIL to be held

Reply:
Is there an assert that can be applied to check? All the uses I've seen below are holding the lock.

Reply:
> This requires the GIL to be held, as do all of the `CPythonAPI` calls in this class. Calling any of these methods without the GIL will likely cause a SEGFAULT or similar

I'm aware of this, but all this publicly visible code is housed under the `Internals` namespace that is considered as reserved for CSnake's own use and currently all the import code invocation is called under the GIL in the generated code.

I hadn't marked this PR as fully ready for review yet but happy to get some early feedback (in fact one of the reasons why I opened it up in draft). I was planning to wrap it up this weekend, post the design notes and also call out things (such as this) as open for debate.

Reply:
This was asserted with 9874b8b513671edcd350057a27da149d797cb5e4, but is now even checked in release builds with afe9a9d8da7f8010028811923f45ecfb5ae40a69.